src/fsharp/FSharp.Core/reflect.fs

```diff
@@ -286,9 +373,22 @@ module internal Impl =
                 match getInstancePropertyReader (typ, "Tag", bindingFlags) with
                 | Some reader -> (fun (obj: obj) -> reader obj :?> int)
                 | None ->
-                    (fun (obj: obj) ->
-                        let m2b = typ.GetMethod("GetTag", BindingFlags.Static ||| bindingFlags, null, [| typ |], null)
-                        m2b.Invoke(null, [|obj|]) :?> int)
+                    let m2b = typ.GetMethod("GetTag", BindingFlags.Static ||| bindingFlags, null, [| typ |], null)
+                    (fun (obj: obj) -> m2b.Invoke(null, [|obj|]) :?> int)
```

Comment:
There's probably no need to look up the method on every invocation. I didn't know how to test this specifically though. When is it ever the case that a DU doesn't have a `Tag` property?

Reply:
A DU always has a tag as far as I know - struct/ref type, single case/multi case all have tag properties.

Reply:
Hmm, then I'm not sure if this branch executes.

Reply:
I think Paul had PR to do tagless DUs a few (lot of) years ago.